### DISCUSSION TITLE
How to predict using multiply equations?

### DISCUSSION THREAD

**Discussion by leelew** [MilesCranmer/PySR#607] [Q&A] [answered]
Hi, 
Thanks for the useful tools! I really enjoy using it.

I want to test the performance of result equations on test dataset. But `model.predict()` only return the predictions from best model. I found it can pass index into `predict`, but I cannot find index of all candidate equations. Is there any methods to predict by all equations shown in hall of frame file?

Regard,
Lu

**Comment by MilesCranmer** [comment, marked as answer]
Thanks!

The equations are stored in `model.equations_` in a pandas dataframe.

For example:

```python
model  # fitted PySRRegressor

for i in range(len(model.equations_)):
    print("Equation", i, "equal to", model.sympy(i))
    print("with predictions")
    pred = model.predict(X, i)
    print(pred)
```

This assumes 1-dimensional `y`. For 2 dimensional you would need to iterate over each component too.

**Reply by leelew** [reply]
Thanks for your useful reply! It works well!
